Sources/StreamChat/Controllers/ChannelWatcherListController/ChatChannelWatcherListController.swift

```diff
@@ -0,0 +1,317 @@
+    ///
+    /// - Note: The delegate can be set directly only if you're **not** using custom extra data types. Due to the current
+    /// limits of Swift and the way it handles protocols with associated types, it's required to use `setDelegate` method
+    /// instead to set the delegate, if you're using custom extra data types.
+    public var delegate: ChatChannelWatcherListControllerDelegate? {
+        get { multicastDelegate.mainDelegate?.wrappedDelegate as? ChatChannelWatcherListControllerDelegate }
+        set { multicastDelegate.mainDelegate = AnyChatChannelWatcherListControllerDelegate(newValue) }
+    }
+}
+
+public extension _ChatChannelWatcherListController {
+    /// Load next set of watchers from backend.
+    ///
+    /// - Parameters:
+    ///   - limit: Limit for page size. Offset is defined automatically by the controller.
+    ///   - completion: The completion. Will be called on a **callbackQueue** when the network request is finished.
+    ///                 If request fails, the completion will be called with an error.
+    func loadNextWatchers(limit: Int = .channelWatchersPageSize, completion: ((Error?) -> Void)? = nil) {
+        var updatedQuery = query
+        updatedQuery.pagination = .init(pageSize: limit, offset: watchers.count)
+        updater.channelWatchers(query: updatedQuery) { error in
+            self.query = updatedQuery
+            self.callback { completion?(error) }
+        }
+    }
+    
+    /// Start watching the channel.
+    /// This function should be called if you want to observe live changes to channel's watchers.
+    ///
+    /// Please check [documentation](https://getstream.io/chat/docs/android/watch_channel/?language=swift) for more information.
+    ///
+    /// - Parameter completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    func startWatching(completion: ((Error?) -> Void)? = nil) {
+        // We have no way to check if channel exists on backend
+        // so we'll let backend return an error if it's not
+        updater.startWatching(cid: query.cid) { error in
+            self.callback {
+                completion?(error)
+            }
+        }
+    }
+    
+    /// Stop watching the channel.
+    /// This function should be called if you want to stop observing live changes to channel's watchers.
+    ///
+    /// Please check [documentation](https://getstream.io/chat/docs/android/watch_channel/?language=swift) for more information.
+    ///
+    /// - Parameter completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    func stopWatching(completion: ((Error?) -> Void)? = nil) {
+        // We have no way to check if channel exists on backend
+        // so we'll let backend return an error if it's not
+        updater.stopWatching(cid: query.cid) { error in
+            self.callback {
+                completion?(error)
+            }
+        }
+    }
```

Comment:
So we want to expose those both on `ChannelController` and here?? What's the reasoning behind this?

Reply:
I guess we can remove these from here... no reason after we introduced the watcherlistcontroller

Reply:
Yep, I think these should live just on `ChannelController`